ln-gateway/src/lib.rs

```diff
@@ -183,22 +181,20 @@ impl LnGateway {
         self.federation_client
             .save_outgoing_payment(contract_account.clone());
 
-        // If this could be an internal payment try to buy the preimage directly. Otherwise or if
-        // this fails try to route the payment via our attached LN node.
-        let preimage_res = payment_params
-            .maybe_internal
-            .then_async(async {
-                self.buy_preimage_internal(&payment_params, &mut rng)
-                    .await
-                    .ok()
-            })
-            .await
-            .flatten()
-            .map(Result::Ok)
-            .unwrap_or_else_async(
-                self.buy_preimage_external(&contract_account.contract.invoice, &payment_params),
-            )
-            .await;
+        let is_internal_payment = payment_params.maybe_internal
+            && self
+                .federation_client
+                .ln_client()
+                .offer_exists(payment_params.payment_hash)
+                .await
+                .unwrap_or(false);
+
+        let preimage_res = if is_internal_payment {
+            self.buy_preimage_internal(&payment_params, &mut rng).await
+        } else {
```

Comment:
This rolls back fallback behavior. Do we want to mark it with an issue so we can reintroduce fallback later?

Reply:
It doesn't because we check if a preimage is for sale before deciding if we should route internally or externally:

```rust
        let is_internal_payment = payment_params.maybe_internal
            && self
                .federation_client
                .ln_client()
                .offer_exists(payment_params.payment_hash)
                .await
                .unwrap_or(false);
```

This wastes one API call in the internal routing case but was better than rearchitecting the entire API client imo.

Reply:
Oh nice. This addresses the private channel case which would look like an internal payment from route hint inspection